Домашнее задание Модуль №9

Ulyanov-VN opened the pull request.

Ulyanov-VN committed:
Домашнее задание Модуль №9

Ulyanov-VN committed:
Домашнее задание Модуль №9

Ulyanov-VN committed:
Merge remote-tracking branch 'origin/dz_m9' into dz_m9

ArturNurtdinov commented on .idea/misc.xml:
Всю папку .idea лучше сразу добавлять в gitignore - в ней хранятся локальные настройки разработчика, специфичные для проекта

ArturNurtdinov commented on app/src/main/AndroidManifest.xml:
```diff
@@ -10,29 +12,32 @@
             android:name=".SearchActivity"
-            android:exported="false" />
+            android:windowSoftInputMode="stateAlwaysVisible|adjustResize" />
+        <!--android:name=".SearchActivity"
+        android:exported="false" />-->
```
Закоментированный код лучше сразу удалять. Хранить закоментированный код в проекте - плохая практика, в случае чего этот код можно восстановить с помощью истории гита

ArturNurtdinov commented on app/src/main/java/com/example/playlistmaker/MediaActivity.kt:
```diff
@@ -30,6 +26,19 @@ class MediaActivity : BaseActivity() {
 
+        // Определяем текущую тему
+        val isNightTheme =
+            when (resources.configuration.uiMode and android.content.res.Configuration.UI_MODE_NIGHT_MASK) {
```
Здесь лучше убрать название пакета, сделав корректный импорт класса или константы

ArturNurtdinov commented on app/src/main/java/com/example/playlistmaker/SearchActivity.kt:
```diff
@@ -1,17 +1,221 @@
+    // Сохранение состояния
+    override fun onSaveInstanceState(outState: Bundle) {
+        super.onSaveInstanceState(outState)
+        outState.putString("SEARCH_TEXT", currentSearchText)
```
Ключи для передачи данных лучше вынести в константу, т.к. они используются в более чем 1 месте. Ещё один плюс вынесения в константы - в будущем легче будет этот экран переиспользовать

ArturNurtdinov commented on app/src/main/java/com/example/playlistmaker/SettingsActivity.kt:
```diff
@@ -1,22 +1,107 @@
+        val prefs = getSharedPreferences("theme_prefs", MODE_PRIVATE)
+        themeSwitch.isChecked = prefs.getBoolean("is_night_mode", false)
+
+        themeSwitch.setOnCheckedChangeListener { _, isChecked ->
+            ThemeHelper.toggleTheme(this, isChecked)
+            recreate() // Пересоздаем активити для применения темы
+        }
```
Андроид умеет это делать самостоятельно при смене конфигурации, пересоздавать активити необязательно

ArturNurtdinov commented on app/src/main/res/drawable/ic_clear.xml:
Для выразительности иконки принято именовать в формате `ic_back_arrow_24`, где `ic` - приставка, обозначающая, что ресурс явялется иконкой, `back_arrow` - простое название, описывающее иконку, а 24 - её размер в dp.

ArturNurtdinov commented on app/src/main/java/com/example/playlistmaker/SearchActivity.kt:
```diff
@@ -1,17 +1,221 @@
+    }*/
+
+    private fun setupSearchField() {
+        // 1. Автоматический фокус и показ клавиатуры
+        searchEditText.setOnClickListener {
+            showKeyboard()
+        }
```
Этот `setOnClickListener` лишний, т.к. EditText из коробки умеет открывать клавиатуру при клике в поле ввода

ArturNurtdinov reviewed (commented):
Привет - очень много закоментированного кода, его необходимо удалить - он затрудняет чтение и ревью кода
1. Отступы между пунктами настроек на экране настроек не соответствует макетам (слишком большой):
![image](https://github.com/user-attachments/assets/a84d75cf-1912-4135-a6ef-234dcbcd1db2)
Макет:
![image](https://github.com/user-attachments/assets/831f7926-073b-4898-a911-c27ef515cccf)
2. Отступы на экране поиска и размер поля ввода не соответствуют макетам - нужно поправить высоту поля ввода, горизонтальные отступы у кнопки "Назад" и горизонтальные отступы у поля ввода - по макетам отступы 16dp:
![image](https://github.com/user-attachments/assets/421f5749-3ccd-44d8-a1aa-96365a6821a6)
Макет:
![image](https://github.com/user-attachments/assets/ddb91dc8-28dd-44bc-aaf4-e97897169618)
3. На экране поиска снизу присутствует лишнее меню BottomNavigationView:
![image](https://github.com/user-attachments/assets/c1259b95-2dde-4918-98e7-bf341a86f3d0)

Ulyanov-VN committed:
Домашнее задание Модуль №9 (Исправление замечаний)
1. Исправил все крически важные замечания (учитывая что нужно было убрать панель навигации, внес изменения в SearchActivity.kt)
2. Прошу прощения, что когда много закоменчено, к следующей домашней работе я удалю все. Часто пишу на работе, мобильный интернет вообще не работает, а на рабочем компьютеры все ресурсы заблокированы
3. Все рекомендации по коду буду исправлять по ходу выполнения следующих домашних заданий, спасибо за комментарии.

ArturNurtdinov commented on app/src/main/res/layout/activity_settings.xml:
```diff
@@ -102,7 +103,7 @@
         android:layout_width="match_parent"
         android:layout_height="64dp"
-        android:layout_marginTop="16dp"
+        android:layout_marginTop="202dp"
```
Так делать не стоит, рекомендуется распологать элементы по вертикали друг за другом, а не через огромный отступ сверху, т.к. в случае увеличенных шрифтов на девайсе все отступы съедут
